test/Provider.coffee

```diff
@@ -161,6 +162,57 @@ describe 'LTI.Provider', () ->
         valid.should.equal true
         done()
 
+    it 'should return true if lti_message_type is ContentItemSelectionRequest', (done) =>
+      req =
+        url: '/test'
+        method: 'POST'
+        connection:
+          encrypted: undefined
+        headers:
+          host: 'localhost'
+        body:
+          lti_message_type: 'ContentItemSelectionRequest'
+          lti_version: 'LTI-1p0'
+          oauth_customer_key: 'key'
+          oauth_signature_method: 'HMAC-SHA1'
+          oauth_timestamp: Math.round(Date.now()/1000)
+          oauth_nonce: Date.now()+Math.random()*100
+
+      #sign the fake request
+      signature = @provider.signer.build_signature(req, req.body, 'secret')
+      req.body.oauth_signature = signature
+
+      @provider.valid_request req, (err, valid) ->
+        should.not.exist err
+        valid.should.equal true
+        done()
+
+    it 'should return false if lti_message_type is ContentItemSelectionRequest and there is a resource_link_id', (done) =>
```

Comment:
I'd be okay with maybe `['field1', 'field2', ... 'field5'].forEach (invalidVield) => { it 'should....' }`